scalafmt-tests/src/test/resources/scala3/FewerBraces.stat

```diff
@@ -1802,3 +1802,312 @@ object a:
+}
+===
+foo
+  .mtd1 { x =>
+    x + 1
+  }
+  .mtd2 { x =>
+    x + 1
+    x + 2
+  }
+  .mtd3 { x =>
+    x + 1
+    x + 2
+    x + 3
+  }
+>>>
+foo
+  .mtd1 { x =>
+    x + 1
+  }
+  .mtd2: x =>
+     x + 1
+     x + 2
+  .mtd3 { x =>
+     x + 1
+     x + 2
+     x + 3
+  }
+<<< rewrite to fewer braces: func in parens
+rewrite.rules = [RedundantBraces]
+rewrite.scala3.removeOptionalBraces = {
+  enabled = yes
+  fewerBracesMinStats = 3
+  fewerBracesMaxStats = 3
+}
+===
+foo
+  .mtd1(
+    x => {
+      x + 1
+    }
+  )
+  .mtd2(
+    x => {
+      x + 1
+      x + 2
+    }
+  )
+  .mtd3(
+    x => {
+      x + 1
+      x + 2
+      x + 3
+    }
+  )
+>>>
+foo
+  .mtd1 { x =>
+    x + 1
+  }
+  .mtd2: x =>
+     x + 1
+     x + 2
+  .mtd3 { x =>
+     x + 1
+     x + 2
+     x + 3
+  }
+<<< rewrite to fewer braces: func in parens and braces
+rewrite.rules = [RedundantBraces, RedundantParens]
+rewrite.scala3.removeOptionalBraces = {
+  enabled = yes
+  fewerBracesMinStats = 3
+  fewerBracesMaxStats = 3
+}
+===
+foo
+  .mtd1(
+    { x =>
+      x + 1
+    }
+  )
+  .mtd2(
+    { x =>
+      x + 1
+      x + 2
+    }
+  )
+  .mtd3(
+    { x =>
+      x + 1
+      x + 2
+      x + 3
+    }
+  )
+>>>
+foo
+  .mtd1 { x =>
+    x + 1
+  }
+  .mtd2: x =>
+     x + 1
+     x + 2
+  .mtd3 { x =>
+     x + 1
+     x + 2
+     x + 3
+  }
+<<< rewrite to fewer braces: partial func
+rewrite.scala3.removeOptionalBraces = {
+  enabled = yes
+  fewerBracesMinStats = 2
+  fewerBracesMaxStats = 2
+}
+===
+foo
+  .mtd1 {
+    case x => x + 1
+  }
+  .mtd2 {
+    case x => x + 1
+    case y => y + 1
+  }
+  .mtd3 {
+    case x => x + 1
+    case y => y + 1
+    case z => z + 1
+  }
+>>>
+foo
+  .mtd1 { case x =>
+    x + 1
+  }
+  .mtd2:
+     case x => x + 1
+     case y => y + 1
+  .mtd3 {
+    case x => x + 1
+    case y => y + 1
+    case z => z + 1
+  }
+<<< rewrite to fewer braces: partial func in parens
+rewrite.rules = [RedundantBraces]
+rewrite.scala3.removeOptionalBraces = {
+  enabled = yes
+  fewerBracesMinStats = 2
+  fewerBracesMaxStats = 2
+}
+===
+foo
+  .mtd1(
+    {
+      case x => x + 1
+    }
+  )
+  .mtd2(
+    {
+      case x => x + 1
+      case y => y + 1
+    }
+  )
+  .mtd3(
+    {
+      case x => x + 1
+      case y => y + 1
+      case z => z + 1
+    }
+  )
+>>>
+foo
+  .mtd1 { case x =>
+    x + 1
+  }
+  .mtd2:
+     case x => x + 1
+     case y => y + 1
+  .mtd3 {
+    case x => x + 1
+    case y => y + 1
+    case z => z + 1
+  }
+<<< rewrite to fewer braces: match
+rewrite.scala3.removeOptionalBraces = {
+  enabled = yes
+  fewerBracesMinStats = 3
+  fewerBracesMaxStats = 3
+}
+===
+foo
+  .mtd1 {
+    bar match {
+      case x => x + 1
+    }
+  }
+  .mtd2 {
+    bar match {
+      case x => x + 1
+      case y => y + 1
+    }
+  }
+  .mtd3 {
+    bar match {
+      case x => x + 1
+      case y => y + 1
+      case z => z + 1
+    }
+  }
+>>>
+foo
+  .mtd1 {
+    bar match
+       case x => x + 1
+  }
+  .mtd2:
```

Comment:
I would say it's only one statement and not count the cases inside.

Reply:
I'll clarify in the codr